File: R/Frederik_ATV.R

#' Frederik Add Two Vectors
#' 
#' Adds two vectors together
#'
#' @param first_vector numeric vector of size n
#' @param second_vector numeric vector of size n
#'
#' @return the vector resulting from adding the two input vectors together. Numeric with size n
#' @export
#'
#' @examples
#' Frederik_ATV(c(1.5,2.5,3.5), c(4.5, 5.5, 6.5))
Frederik_ATV = function(first_vector, second_vector) {
  result.list = .C("Frederik_ATV_interface",
              as.double(first_vector),
              as.double(second_vector),
              as.integer(length(first_vector)),
              as.integer(length(second_vector)),
              result_vector = double(as.integer(length(first_vector))),
              PACKAGE="FrederikClusteringAlgorithms")
  
  return(result.list$result_vector)
  
}

File: src/Frederik_ATV.cpp
#include "Frederik_ATV.h"

int Frederik_ATV(
    const double *first_vector, //first vector, with length n_length
    const double *second_vector, //second vector, with length n_length
    const int first_length, //length of the first vector
    const int second_length, //length of the second vector
    double *resulting_vector //the resulting vector, length n_length
) {
  
  if (!(first_length > 0) || !(second_length > 0)) {
    return ERROR_VECTOR_LENGTH_MUST_BE_GREATER_THAN_ZERO;
  }
  if (first_length != second_length) {
    return ERROR_VECTORS_MUST_HAVE_SAME_LENGTH;
  }
  
  for(int i=0; i<first_length; i++) {
    resulting_vector[i] =
      first_vector[i] + second_vector[i];
  }
  
  return 0;
}